Leak of id<MTLArgumentEncoder> in rx::(anonymous)::InitArgumentBufferEncoder()

ddkilzer opened the pull request:
#### 06f2309860842383d90df8482abdb072d91c4f5d
<pre>
Leak of id&lt;MTLArgumentEncoder&gt; in rx::(anonymous)::InitArgumentBufferEncoder()
<a href="https://bugs.webkit.org/show_bug.cgi?id=244008">https://bugs.webkit.org/show_bug.cgi?id=244008</a>
&lt;rdar://98747531&gt;

Reviewed by Kimmo Kinnunen and Dean Jackson.

* Source/ThirdParty/ANGLE/changes.diff: Update.
* Source/ThirdParty/ANGLE/src/libANGLE/renderer/metal/ProgramMtl.mm:
(rx::(anonymous)::InitArgumentBufferEncoder):
- Use mtl::adoptObjCObj() to fix the leak since
  -newArgumentEncoderWithBufferIndex: returns a +1 retained object.

Canonical link: <a href="https://commits.webkit.org/253525@main">https://commits.webkit.org/253525@main</a>
</pre>

kkinnunen-apple reviewed (approved):
If you want, you can also run Tools/Scripts/update-angle --regenerate-changes-diff after your commit and amend with  the changes.diff changes.
I can submit this upstream if you want.

Commit:
Leak of id<MTLArgumentEncoder> in rx::(anonymous)::InitArgumentBufferEncoder()
https://bugs.webkit.org/show_bug.cgi?id=244008
<rdar://98747531>

Reviewed by Kimmo Kinnunen and Dean Jackson.

* Source/ThirdParty/ANGLE/changes.diff: Update.
* Source/ThirdParty/ANGLE/src/libANGLE/renderer/metal/ProgramMtl.mm:
(rx::(anonymous)::InitArgumentBufferEncoder):
- Use mtl::adoptObjCObj() to fix the leak since
  -newArgumentEncoderWithBufferIndex: returns a +1 retained object.

Canonical link: https://commits.webkit.org/253525@main
